santoku/utils/url_handler.py

```diff
@@ -212,152 +344,41 @@ def contains_ip(cls, url: str) -> bool:
-
-                if urlparse(url).scheme:
-                    return [""]
-
-                return [url]
-
-        raise InvalidURLError(f"The {url} URL does not contain domain or suffix")
-        # We comment this code block out until we are sure of what to do
-        # try:
-        #     res = tld.get_tld(url, fix_protocol=True, as_object=True)
-        # except (tld.exceptions.TldDomainNotFound, tld.exceptions.TldBadUrl) as error:
-        #     # get_tld raises an exception when the top level domain (tld) is unknown
-        #     # For example, we might find an unknown tld if someone uses ".devel" during development
-        #     # The code below is an attempt to "clean" the domain by doing
-        #     # - Remove http:// and https://
-        #     # - Split by "/" and return position 0 of the list
-        #     # - Split by "?" and return position 0
-        #     # - Split by "#" and return position 0
-        #     parsed_url = url.replace("http://", "").replace("https://", "").replace("//", "")
-        #     parsed_url = parsed_url.split("/")[0].split("?")[0].split("#")[0]
-        #     return [parsed_url]
-        # exploded_subdomains = [res.fld]
-
-    @classmethod
-    def get_path(cls, url: str, raise_exception_if_invalid_url: bool = True) -> str:
-        """
-        Given a URL, return the path.
-
-        When the given URL is invalid, if `raise_exception_if_invalid_url` is set to `True`,
-        `InvalidURLError`exception will be raised. If the url does not contain a scheme or a domain,
-        an empty string will be returned instead.
-
-        Parameters
-        ----------
-        url : str
-            The URL to get the path from.
-
-        raise_exception_if_invalid_url : bool, Optional
-            We consider as invalid those URLs in which some particles are missing in the fully
-            qualified domain name. Set to `True` by default.
-
-        Returns
-        -------
-        str
-            The path of the URL or an empty string, following the aforementioned criteria.
-
-        Raises
-        ------
-        InvalidURLError
-            If `url` is invalid.
-
-        Example
-        -------
-        'https://example.com/path/' -> '/path/'
-
-        """
-        res = tldextract.extract(url)
-
-        # When URL is invalid
-        if not res.domain or not res.suffix:
-            if raise_exception_if_invalid_url:
-                raise InvalidURLError(f"The {url} URL does not contain domain or suffix")
-
-        parsed_url = urlparse(url)
-        if parsed_url.scheme or parsed_url.netloc:
-            return parsed_url.path
-        else:
-            return ""
+        exploded_domains = []
+        # Add 1 to include all subdomains
+        for num_subdomains in range(len(self.subdomains) + 1):
+            partial_domain = self.get_partial_domain(num_subdomains=num_subdomains)
+            if partial_domain:
+                exploded_domains.append(partial_domain)
```

Comment:
It is correct, but not the most efficient since `get_partial_domain` is being called multiple times. We have refactored it to only call it once and then do successive splits on "." to obtain the different `partial_subdomain`.